src/main/java/com/three_tech_solutions/slot_app/services/implementations/PaymentServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.three_tech_solutions.slot_app.services.implementations;
+
+import com.three_tech_solutions.slot_app.data.models.Payment;
+import com.three_tech_solutions.slot_app.data.models.Student;
+import com.three_tech_solutions.slot_app.data.repositories.PaymentRepository;
+import com.three_tech_solutions.slot_app.services.interfaces.PaymentService;
+import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+
+@AllArgsConstructor
+@Service
+public class PaymentServiceImpl implements PaymentService {
+    private final PaymentRepository paymentRepository;
+
+    @Override
+    @Transactional
+    public Payment createPayment(Student student, double amount) {
+        int newPaymentNumber = getNextPaymentNumber();
+
+        Payment payment = new Payment(
+                newPaymentNumber,
```

Comment:
Acá podrías directamente llamar al método dentro de la llamada al constructor


```suggestion
        Payment payment = new Payment(
                getNextPaymentNumber(),
```